src/main/scala/com/github/unchama/seichiassist/menus/home/HomeMenu.scala

```diff
@@ -0,0 +1,332 @@
+import com.github.unchama.seichiassist.concurrent.PluginExecutionContexts.onMainThread
+import com.github.unchama.seichiassist.menus.CommonButtons
+import com.github.unchama.seichiassist.menus.stickmenu.FirstPage
+import com.github.unchama.seichiassist.subsystems.home.HomeReadAPI
+import com.github.unchama.seichiassist.subsystems.home.domain.HomeId
+import com.github.unchama.seichiassist.{ManagedWorld, SkullOwners}
+import com.github.unchama.targetedeffect._
+import com.github.unchama.targetedeffect.player.PlayerEffects._
+import com.github.unchama.targetedeffect.player.{CommandEffect, FocusedSoundEffect}
+import eu.timepit.refined.auto._
+import org.bukkit.ChatColor._
+import org.bukkit.entity.Player
+import org.bukkit.{Material, Sound}
+
+object HomeMenu {
+
+  class Environment(
+    implicit val ioCanOpenConfirmationMenu: IO CanOpen HomeChangeConfirmationMenu,
+    implicit val ioCanOpenFirstPage: IO CanOpen FirstPage.type,
+    implicit val ioCanOpenHome: IO CanOpen HomeMenu,
+    val ioCanOpenHomeRemoveConfirmationMenu: IO CanOpen HomeRemoveConfirmationMenu,
+    implicit val homeReadAPI: HomeReadAPI[IO],
+    implicit val asyncShift: NonServerThreadContextShift[IO]
+  )
+
+}
+case class HomeMenu(pageIndex: Int = 0) extends Menu {
+
+  private val pageIndexMax = 0 max (HomeId.maxNumber - 1) / 9
+  override type Environment = HomeMenu.Environment
+
+  override val frame: MenuFrame =
+    MenuFrame(5.chestRows, s"$DARK_PURPLE${BOLD}ホームメニュー ${pageIndex + 1}/${pageIndexMax + 1}")
+
+  override def computeMenuLayout(
+    player: Player
+  )(implicit environment: Environment): IO[MenuSlotLayout] = {
+    import eu.timepit.refined._
+    import eu.timepit.refined.auto._
+    import eu.timepit.refined.numeric._
+
+    val buttonComputations = HomeMenuButtonComputations(player)
+    import buttonComputations._
+
+    val homePointPart = for {
+      homeNumber <- 1 + (9 * pageIndex) to HomeId.maxNumber - 9 * (pageIndexMax - pageIndex)
+    } yield {
+      val column = refineV[Interval.ClosedOpen[0, 9]](homeNumber - 9 * pageIndex - 1)
+      column.fold(
+        _ => throw new RuntimeException("This branch should not be reached."),
+        value =>
+          Map(
+            ChestSlotRef(0, value) -> ConstantButtons.warpToHomePointButton(homeNumber),
+            ChestSlotRef(2, value) -> ConstantButtons.setHomeButton(homeNumber),
+            ChestSlotRef(3, value) -> ConstantButtons.removeHomeButton(homeNumber)
+          )
+      )
+    }
+
+    import environment._
+    // 5スロット目のページ遷移メニュー定義
+    val paginationPartMap = {
+      val stickButtonMap = Map(ChestSlotRef(4, 0) -> CommonButtons.openStickMenu)
+      val prevButtonMap = {
+        MapExtra.when(pageIndex >= 1)(
+          Map(
+            ChestSlotRef(4, 7) -> CommonButtons.transferButton(
+              new SkullItemStackBuilder(SkullOwners.MHF_ArrowLeft),
+              s"${pageIndex}ページ目へ",
+              HomeMenu(pageIndex - 1)
+            )
+          )
+        )
+      }
```

Comment:
```suggestion
      val prevButtonMap = 
        MapExtra.when(pageIndex >= 1)(
          Map(
            ChestSlotRef(4, 7) -> CommonButtons.transferButton(
              new SkullItemStackBuilder(SkullOwners.MHF_ArrowLeft),
              s"${pageIndex}ページ目へ",
              HomeMenu(pageIndex - 1)
            )
          )
        )
```
カッコがいらないのと、`transferButton`を変数に切り出すとわかりやすいと思います。